### PR TITLE
DENG-3367 add partnership udf logic to views

### PR DESCRIPTION
**Pull request by kwindau**
This PR adds the new "partnership UDFs" to the following 3 views: 

- moz-fx-data-marketing-prod.ga.www_site_metrics_summary
- mozdata.telemetry.clients_first_seen
- mozdata.telemetry.clients_first_seen_28_days

It also adds the 6 stub tests for these 6 UDFs used in these 3 views since the actual UDF logic lives in private-bigquery-etl which the stage testing environment doesn't have access to.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3369)

### CODE REVIEW

File: sql/moz-fx-data-marketing-prod/ga/www_site_metrics_summary/view.sql [hunk 1]
```diff
@@ -2,8 +2,15 @@ CREATE OR REPLACE VIEW
...
+  `moz-fx-data-shared-prod`.udf.funnel_derived_ga_metrics(
+    device_category,
+    browser,
+    operating_system
+  ) AS funnel_derived,
+  `moz-fx-data-shared-prod`.udf.distribution_model_ga_metrics() AS distribution_model,
+  `moz-fx-data-shared-prod`.udf.partner_org_ga_metrics() AS partner_org
 FROM
-  `moz-fx-data-marketing-prod.ga_derived.www_site_metrics_summary_v2`
+  `moz-fx-data-marketing-prod.ga_derived.www_site_metrics_summary_v2` a
```
Comment (BenWu) [inline sql/moz-fx-data-marketing-prod/ga/www_site_metrics_summary/view.sql:14]: Why was an alias added here?

File: sql/moz-fx-data-shared-prod/udf/distribution_model_ga_metrics/udf.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
+This is a stub implementation for use with tests in this repo
+Real implementation is in private-bigquery-etl
+*/
+CREATE OR REPLACE FUNCTION udf.distribution_model_ga_metrics()
+RETURNS STRING AS (
+  (SELECT 'helloworld' AS distribution_model)
+);
```
Comment (BenWu) [inline sql/moz-fx-data-shared-prod/udf/distribution_model_ga_metrics/udf.sql:8]: Is there a difference between this and
```suggestion
RETURNS STRING AS (
  'helloworld'
);
```

**Review by BenWu — APPROVED** [review]
LGTM, I just have a couple questions trivial questions

File: sql/moz-fx-data-marketing-prod/ga/www_site_metrics_summary/view.sql (hunk 1 above)
Comment (kwindau) [reply sql/moz-fx-data-marketing-prod/ga/www_site_metrics_summary/view.sql:14]: Out of habit, I can take it out XD

**Pull request merged**